cumulusci/core/config/project_config.py

```diff
@@ -498,8 +507,62 @@ def get_repo_from_url(self, url):
 
         return repo
 
-    def get_ref_for_dependency(self, repo, dependency, include_beta=None):
-        release = None
+    def find_matching_2gp_release(self, remote_repo, context_2gp):
+        # To allow us to locate release branches on the remote repo, we need to know its feature branch prefix.
+        # We'll use the cumulusci.yml file from HEAD on the main branch to determine this.
+
+        release_id = get_release_identifier(
+            self.repo_branch, self.project__git__prefix_feature
+        )
+        remote_branch_prefix = None
+        contents = remote_repo.file_contents(
+            "cumulusci.yml",
+            ref=remote_repo.branch(remote_repo.default_branch).commit.sha,
+        )
+        head_cumulusci_yml = cci_safe_load(
+            io.StringIO(contents.decoded.decode("utf-8"))
+        )
+        remote_branch_prefix = (
+            head_cumulusci_yml.get("project", {})
+            .get("git", {})
+            .get("prefix_feature", "feature/")
+        )
+
+        if not remote_branch_prefix:
+            return None
+
+        remote_matching_branch = construct_release_branch(
+            remote_branch_prefix, release_id
+        )
+
+        # Check the most recent commit on this release branch looking for a 2GP package to use
+        # TODO: Should we attempt to scan backwards to address race conditions?
+        # Or look for a commit status for an in-progress package upload, and poll?
```

Comment:
I think we probably want the former, so that a new commit on the remote release branch doesn't make it impossible to deploy downstream packages using this option until the package has been created. But we can do it without first as an MVP.

I think the value of polling is probably low since these skip-validation uploads are quite quick.